firebaseFunctions/src/types.ts

```diff
@@ -0,0 +1,112 @@
+interface IProposal {
+  title: string,
+  summary: string,
+  description: string;
+}
+
+enum Platforms {
+  "Independent",
+  "DAOstack",
+  "Moloch",
+  "OpenLaw",
+  "Aragon",
+  "Colony",
+  "Compound Governance",
+  "Snapshot",
+  "Gnosis Safe / Snapshot",
+  "Substrate",
+}
+
+interface IToken {
+  address: string,
+
+  name: string,
+  symbol: string,
+  decimals: number,
+  logoURI: string,
+
+  amount: string
+  instantTransferAmount: string
+  vestedTransferAmount: string
+  vestedFor: number
+  cliffOf: number
+}
+
+interface ISocialMedia {
+  name: string,
+  url: string,
+}
+
+interface IDAO {
+  name: string;
+  treasury_address: string;
+  logoURI: string;
+  social_medias: Array<ISocialMedia>;
+  representatives: Array<{address: string}>;
+  id?: string;
+  tokens?: Array<IToken>;
+  platform?: Platforms;
+}
+
+interface IProposalLead {
+  address: string,
+  email?: string;
+  // dao?: IDAO /* Deprecated: Proposal lead does not need to be part of the a DAO */
+}
+
+interface IClause {
+  id: string,
+  text: string,
+}
+
+interface ITerms {
+  clauses: Array<IClause>,
+}
+
+export interface IDealRegistrationTokenSwap {
```

Comment:
Maybe there is a way to have a "Shared" directory where we can store interfaces used both by frontend and backend?